controllers/remote/cluster_cache.go

```diff
@@ -159,7 +170,7 @@ func (m *ClusterCacheTracker) Watch(ctx context.Context, input WatchInput) error
 	wi := watchInfo{
 		watcher:      input.Watcher,
 		kind:         input.Kind,
```

Comment:
Kind here is also an interface and we have no control over what it contains, it's technically possible to pass `&corev1.Node{}` and `&corev1.Node{ObjectMeta:{Name:"x"}}` in here. What about changing this to be GVK instead? We can use the runtime.Scheme from the Manager when we create the tracker to safely determine it.